src/search/pruning_method.h

```diff
@@ -15,22 +16,30 @@ namespace limited_pruning {
 class LimitedPruning;
 }
 
+namespace options {
+class OptionParser;
+class Options;
+}
+
 class PruningMethod {
     utils::Timer timer;
     friend class limited_pruning::LimitedPruning;
 
     virtual void prune(
         const State &state, std::vector<OperatorID> &op_ids) = 0;
 protected:
+    mutable utils::LogProxy log;
```

Comment:
I assume this does have to be mutable? It's not pretty.

Reply:
Yes, unfortunately, because output streams are modified when used.

Reply:
If we think it is OK to do output on const logs, it would probably be good to encapsulate this inside the LogProxy class. But that's something for another issue then.